cloud/vsphere/machineactuator.go

```diff
@@ -231,29 +273,48 @@ func (vc *VsphereClient) Create(cluster *clusterv1.Cluster, machine *clusterv1.M
 		return errors.New(fmt.Sprintf("Could not write startup script %s", err))
 	}
 
+	return nil
+}
+
+func (vc *VsphereClient) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	config, err := vc.machineproviderconfig(machine.Spec.ProviderConfig)
+	if err != nil {
+		return vc.handleMachineError(machine, apierrors.InvalidMachineConfiguration(
+			"Cannot unmarshal providerConfig field: %v", err))
+	}
+
+	clusterConfig, err := vc.clusterproviderconfig(cluster.Spec.ProviderConfig)
+	if err != nil {
+		return err
+	}
+
+	if verr := vc.validateMachine(machine, config); verr != nil {
```

Comment:
Is there a reason to use a new error (verr) instead of using the previously declared err?

Reply:
No -- legacy code.